pkg/services/sqlstore/permissions/dashboard.go

```diff
@@ -365,3 +420,24 @@ func nestedFoldersSelectors(permSelector string, permSelectorArgs []interface{},
 
 	return strings.Join(wheres, ") OR "), args
 }
+
+func parseStringSliceFromInterfaceSlice(slice []interface{}) []string {
+	result := make([]string, 0, len(slice))
+	for _, s := range slice {
+		result = append(result, s.(string))
+	}
+	return result
+}
+
+func parseArguments(actions []string, user *user.SignedInUser, scopePrefix string) []interface{} {
```

Comment:
I think this logic would list all the dashboards that user has **any** of the required permissions on, while the old logic was listing dashboards that user has **all** of the required permissions on. Or am I misunderstanding something?

Reply:
Thanks! I investigated this and you are right! I pushed a fix + added tests for this scenario!